src/core/compile.ts

```diff
@@ -58,5 +60,5 @@ export function compile(
     }
 
     // Make HiGlass models for individual tracks
-    createHiGlassModels(JSON.parse(JSON.stringify(specCopy)), trackInfos, setHg, theme);
+    createHiGlassModels(JSON.parse(JSON.stringify(specCopy)), trackInfos, callback, theme);
```

Comment:
```suggestion
    createHiGlassModels(structuredClone(specCopy), trackInfos, callback, theme);
```

Reply:
For separate PR/consideration... but this cloning of the spec makes me assume the function is mutating the object in place. I think it would be preferred to have have the function avoid mutation of its arguments and instead clone the object internally and _return_ a new object (whether a clone or not).

Reply:
Right, that I think makes more sense.